### PR TITLE
Promote div and / in the same way

### PR DESCRIPTION
**Pull request by hersle**
The current operation promotion falls back to `0 / 1` for `/`, but `div(0, 0)` for `div`, giving the `DivideError` in #240.
This fixes that.

### CODE REVIEW

**Comment by odow** [comment]
Nice! Could you add a new test for this to the bottom of https://github.com/jump-dev/MutableArithmetics.jl/blob/master/test/interface.jl?

File: src/interface.jl [hunk 1]
```diff
@@ -33,16 +33,16 @@ _instantiate(::Type{S}) where {S<:Irrational} = S()
 _instantiate_zero(::Type{S}) where {S<:AbstractIrrational} = _instantiate(S)
 _instantiate_oneunit(::Type{S}) where {S<:AbstractIrrational} = _instantiate(S)
 
+# Julia v1.0.x has trouble with inference with the `Vararg` method, see
+# https://travis-ci.org/jump-dev/JuMP.jl/jobs/617606373
 function promote_operation_fallback(
-    ::typeof(/),
+    op::Union{typeof(/), typeof(div)},
```
Comment (odow) [inline src/interface.jl:39]:
```suggestion
    op::Union{typeof(/),typeof(div)},
```

**Comment by hersle** [comment]
I added a test 😃

File: test/interface.jl [hunk 2]
```diff
@@ -178,3 +178,11 @@ end
...
     @test MA.operate_to!!(T(6), abs, T(-7)) == 7
 end
+
+@testset "Error-free mutability (issue #240)" begin
+    for op in (+, -, *, /, div)
+        for T in (Float64, BigFloat, Int, BigInt, Rational{Int}, Rational{BigInt})
+            @test_nowarn MA.mutability(T, op, T, T) # should run without error
+        end
+    end
+end
```
Comment (odow) [inline test/interface.jl:188]:
```suggestion
end

```

**Comment by odow** [comment]
The formatter is very particular :smile: but it does help unify the code across all of `jump-dev`.

**Comment by hersle** [comment]
That's alright :smile:

**Pull request merged**